frontend/components/SetupPage/Create/SetupEoaFunding/useBeforeBridgeFunds.ts

```diff
@@ -0,0 +1,54 @@
+import { useCallback } from 'react';
+
+import { SERVICE_TEMPLATES } from '@/constants/serviceTemplates';
+import { AgentType } from '@/enums/Agent';
+import { useServices } from '@/hooks/useServices';
+import { useStakingProgram } from '@/hooks/useStakingProgram';
+import { onDummyServiceCreation } from '@/utils/service';
+
+/**
+ * Hook to run actions before bridging funds screen.
+ * For predict agent, it creates a dummy service & then navigates to the bridge onboarding.
+ */
+export const useBeforeBridgeFunds = () => {
+  const { defaultStakingProgramId } = useStakingProgram();
+  const {
+    selectedAgentType,
+    selectedService,
+    refetch: refetchServices,
+  } = useServices();
+
+  const serviceTemplate = SERVICE_TEMPLATES.find(
+    (template) => template.agentType === selectedAgentType,
+  );
+
+  return useCallback(async () => {
+    // For predict agent: create a dummy service and then navigate to bridge onboarding
+    if (selectedAgentType === AgentType.PredictTrader) {
```

Comment:
can we just check if the service is not created and create it? this is more logical rather than checking the exact agent and ensures the code is agent agnostic. otherwise we risk to forget about this when adding a new agent that doesn't require a setup